### DISCUSSION TITLE
Python scripts hosting

### DISCUSSION THREAD

**Discussion by vkwd** [community/community#21470] [Other Feature Feedback, Questions, & Ideas]
<p>I’m a newbie in programming. I have created a python script that calculates pension of retirees, where the user inputs the basic pension and the output is gross pension. the script runs successfully in vscode, widows terminal.<br>
A similar calculator in javascript that I made, I can upload on blogger and a website. But I do not know how to upload the python file  on blogger or website.<br>
Searches on the internet revealed that I need a frame work like Django/flask creating the calculator.<br>
My query is; if python scripts cannot be shared on a blogger /website then what is the way left if the users mostly use them.<br>
2. Even if I create a calculator in Django, will it be possible to upload on a blogger or website .<br>
3. what are the alternatives left for me ?<br>
thanks</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="vkwd" data-post="1" data-topic="136806">
<div class="title">
<div class="quote-controls"></div>
 vkwd:</div>
<blockquote>
<p>My query is; if python scripts cannot be shared on a blogger /website then what is the way left if the users mostly use them.</p>
</blockquote>
</aside>
<p>Installing them locally. Depending on the complexity and target user group you might just offer the script for download, <a href="https://packaging.python.org/guides/distributing-packages-using-setuptools/">create a pip package</a>, or set up installers. Also depending on the target use you may or may not want to add a GUI.</p>
<aside class="quote no-group" data-username="vkwd" data-post="1" data-topic="136806">
<div class="title">
<div class="quote-controls"></div>
 vkwd:</div>
<blockquote>
<ol start="2">
<li>Even if I create a calculator in Django, will it be possible to upload on a blogger or website .</li>
</ol>
</blockquote>
</aside>
<p>Frameworks like Django or Flask need to <em>run</em> on the server. When a user makes a request they process the input and send a suitable response. Depending on your architecture you may want to generate a whole HTML site (maybe with forms), or serve static HTML and Javascript that then talk to the web application using a REST interface for the dynamic part.</p>
